### PR TITLE
Set up Nginx 

### PR DESCRIPTION
**Pull request by SyedTahaA**
- backend proxy
- serve frontend 
- Added AWS load balancer

related: #11

### CODE REVIEW

File: Buddies.Web/package.json [hunk 1]
```diff
@@ -3,7 +3,7 @@
   "private": true,
   "scripts": {
     "dev": "next dev",
-    "build": "next build",
+    "build": "next build && next export",
```
Comment (andy9775) [inline Buddies.Web/package.json:6]:
```suggestion
    "build": "next build",
    "export": "next export",
```

File: appspec.yml [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    - location: scripts/startapp.sh
+      timeout: 300
+      runas: root
+    - location: scripts/startapi.sh
+      timeout: 300
+      runas: root
+  ApplicationStop:
+    - location: scripts/stopapi.sh
+      timeout: 300
+      runas: root
```
Comment (andy9775) [inline appspec.yml:20]: This file is ci/cd specific and it should only reside in that pr. This pr should only contain nginx related config.

File: configuration/buildspec.yml [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+      - echo Post Build started on `date`
+      - dotnet publish -c Release -r linux-x64 --self-contained false Buddies.API/Buddies.API.csproj
+artifacts:
+  files:
+    - Buddies.Web/out/**/*
+    - Buddies.API/bin/Release/net6.0/linux-x64/publish/**/*
+    - appspec.yml
+    - scripts/startapp.sh
+    - scripts/startapi.sh
+    - scripts/stopapi.sh
```
Comment (andy9775) [inline configuration/buildspec.yml:36]: Same here.

File: docs/DEPLOY.md [hunk 4]
```diff
@@ -1 +1,116 @@
...
+  - Choose CodePipeline default for Output artifact format.
+  - Choose *AWS CodeBuild* for Build provider
+      - Create a new build project
+      - Choose *Managed image* for Environment image and *Amazon Linux 2* for Operating System.
+      - Select *Standard* Runtime and *aws/codebuild/amazonlinux2-x86_64-standard:3.0* for image.
+      - Choose *aws/codebuild/amazonlinux2-x86_64-standard:3.0-21.10.15* for image version and *Linux* for Environment type.
+      - Select *New service role* for Service role.
+      - Create and add **buildspec.yml** to github repo and put its path for the Build specifications name.
+      - Continue to CodePipeline
+  - For Deploy provider, select *AWS CodeDeploy* and choose the previously created application and deployment group.
```
Comment (andy9775) [inline docs/DEPLOY.md:116]: Again, this shouldn't have any nginx configuration.

File: nginx/react.conf [hunk 5]
```diff
@@ -0,0 +1,23 @@
+server {
+    listen 80; 
+    listen [::]:80;
+    server_name _;
```
Comment (andy9775) [inline nginx/react.conf:4]:
```suggestion
    server_name localhost;
```

**Review by andy9775 — CHANGES_REQUESTED** [review]
Make sure that this is only nginx stuff

**Comment by andy9775** [comment]
Also, add one other person as a reviewer in case you forgot.

File: appspec.yml (hunk 2 above)
Comment (SyedTahaA) [reply appspec.yml:20]: should i make a new branch and add the nginx stuff there, cause i branched this branch from the ci/cd branch so a lot of the non-nginx related files got included.

Comment (andy9775) [reply appspec.yml:20]: This should have been branched off of main and only included the nginx config.

**Comment by andy9775** [comment]
We can close this I'm assuming?

**Pull request closed without merging**